library/ssl_tls13_client.c

```diff
@@ -2203,6 +2202,20 @@ static int ssl_tls13_process_encrypted_extensions(mbedtls_ssl_context *ssl)
     }
 #endif
 
+    /*
+     * When early_data extension is enabled and sent in ClientHello, the client
+     * does not know if the server will accept early data and select the first
+     * proposed pre-shared key with a ciphersuite that is different from the
+     * ciphersuite associated to the selected pre-shared key. To address
+     * aforementioned case, we do associated verification when parsing
+     * early_data ext in EncryptedExtensions. Therefore we have to assign
+     * the ciphersuite in current handshake to session_negotiate later than
+     * the associated verification. This won't impact decryption of
+     * EncryptedExtensions since we compute handshake keys by the ciphersuite
+     * in handshake not via the one in session_negotiate.
+     */
+    ssl->session_negotiate->ciphersuite = handshake->ciphersuite_info->id;
```

Comment:
```suggestion
    /*
     * Move `session_negotiate->ciphersuite` assign here which after early data cipher suite check.(describe the target)
     *
     * `session_negotiate->ciphersuite` is ciphersuite we proposed in client hello and it is not used for computing handshake_transform. And early data need ciphersuite check. 
     * 
     */
    ssl->session_negotiate->ciphersuite = handshake->ciphersuite_info->id;
```

Reply:
Here we should put the action and why we can do the action. 

above early data cipher check, add reason why we should do the check.